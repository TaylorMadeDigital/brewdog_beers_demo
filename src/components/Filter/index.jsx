import React, { useState, useEffect } from 'react';
import { useDispatch } from "react-redux";
import Dropdown from '../Dropdown';
import s from './index.scss';
import { FILTER_OPTIONS } from '../../consts/filterOptions';
import { CONTENT } from '../../consts/content';

const FilterComponent = () => {

  const [filter, setFilter] = useState("");
  const dispatch = useDispatch();

  const handleChange = (newValue) => setFilter(newValue);

  useEffect(() => {
    function updateFilter() {
      dispatch({ type: 'FILTER_BY', payload: filter });
    } 
    updateFilter();
  }, [dispatch, filter]);

  return (
    <div className={s.root}>
      <label htmlFor="filterDropdown">{CONTENT.filter.title}</label>
      <Dropdown id="filterDropdown" value={filter} onChange={handleChange} options={FILTER_OPTIONS} />
    </div>
  );
};

export default FilterComponent;
