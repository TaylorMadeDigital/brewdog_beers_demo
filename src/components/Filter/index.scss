@import '../../assets/scss/variables';

.root {
    background: $toolbar-bg-color;
    padding: $space-base $space-m;

    label {
      color: $light-text-color;
      font-weight: bold;
      margin: 0 $space-base 0 0;
    }
}