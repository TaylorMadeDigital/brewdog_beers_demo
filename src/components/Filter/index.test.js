import React from 'react';
import { render, cleanup } from "@testing-library/react";
import Filter from './index';
import { createStore } from 'redux'
import { Provider } from 'react-redux';

describe('<Filter />', () => {

  afterEach(cleanup);

  it('snapshot: renders default state', () => {

    const store = createStore(() => ({}))
    const { asFragment } = render(<Provider store={store}><Filter /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
