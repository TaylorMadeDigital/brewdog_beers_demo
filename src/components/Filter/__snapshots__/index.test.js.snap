// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> snapshot: renders default state 1`] = `
<DocumentFragment>
  <div>
    <label
      for="filterDropdown"
    >
      Order by:
    </label>
    <select
      data-testid="dropdown"
      id="filterDropdown"
    >
      <option>
        - Select an option -
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="first_brewed"
      >
        Date first brewed
      </option>
      <option
        value="abv"
      >
        ABV
      </option>
      <option
        value="ibu"
      >
        IBU
      </option>
    </select>
  </div>
</DocumentFragment>
`;
